articles/gosagawa.re

```diff
@@ -0,0 +1,61 @@
+= Goのエラーハンドリングを深掘りする
+
+== 概要
+
+アプリケーションを実装するにあたり、エラーハンドリングは非常に重要です。しかし、エラー管理がデフォルトでは非常にシンプルです。カスタマイズをしなければ不便であるところは欠点でありつつもの、逆にしっかりとカスタマイズをすれば非常に強力なツールとなりえます。この記事ではpkg/errorsライブラリや、GopherCon2019のセッション"Handling go errors",そしてAthensプロジェクトでのエラーハンドリングの
+実例を通して、実用にみあったエラーハンドリング処理の書き方をお伝えできればと思っています。
+
+== エラーハンドリングで満たしたい要件
+
+Goのエラーハンドリングの良いところでもあり大変なところでもあるのは、カスタマイズがしやすく設計が必要であるという事だと思います。
+何もせず、そのままreturn errを繰り返すだけだとエラーのテキスト情報のみしか受け渡す事ができません。
+
+では、実際の業務でエラーハンドリングに求められる要件はどのような事があるでしょうか？
```

Comment:
**[textlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
Disallow to use "？". (ja-technical-writing/no-exclamation-question-mark)